### ISSUE TITLE
TypeScript coding guidelines say to not prefix interface with I

### ISSUE THREAD

**Issue by bdparrish** [palantir/tslint#3265]
### Bug Report

- __TSLint version__: 3.15.1
- __TypeScript version__: 2.5.3
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
// code snippet
interface TypeConfig { }
```

#### Actual behavior
[tslint] interface name must start with a capitalized I (interface-name)

Microsoft suggests interface names to not  be prefixed with an I - https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#names

**Comment by ajafff** [comment]
Note that these are the guidelines for writing the typescript compiler itself. It's not meant to be a style guide for any other typescript project out there.
If you want to adopt this style guide for your projects, you can just change the rule's config in your tslint.json to `"interface-name" : [true, "never"]`

**Comment by kachkaev** [comment]
50 cent from a passer-by:  
`"interface-name" : [true, "never-prefix"]` instead of  
`"interface-name" : [true, "never"]`
🙌 

[rule docs](https://palantir.github.io/tslint/rules/interface-name/)

**Comment by mraible** [comment]
I tried turning off the "I" prefix by adding interface-name to the default `tslint.json` that's generated by Create React App. However, it doesn't seem to work. Maybe I'm putting it in the wrong location?

```json
{
  "extends": [
    "tslint:recommended",
    "tslint-react",
    "tslint-config-prettier"
  ],
  "linterOptions": {
    "exclude": [
      "config/**/*.js",
      "node_modules/**/*.ts"
    ]
  },
  "interface-name" : [true, "never-prefix"]
}
```

**Comment by mraible** [comment]
Nevermind, I figured it out:

```json
{
  "extends": [
    "tslint:recommended",
    "tslint-react",
    "tslint-config-prettier"
  ],
  "linterOptions": {
    "exclude": [
      "config/**/*.js",
      "node_modules/**/*.ts"
    ]
  },
  "rules": {
    "interface-name" : [true, "never-prefix"]
  }
}
```

**Comment by vasilev-alex** [comment]
For some reason `"interface-name": [true, "never"]` doesn't work for me. But `"interface-name": false` works fine

**Comment by aaronjameslang** [comment]
@vasilev-alex `"interface-name": [true, "never-prefix"]` empasis on never-**prefix**

**Comment by offg777** [comment]
I'm still getting `"interface name must start with a capitalized I"`

this is my tslint.json:
```
{
  "rules": {
    "interface-name": [
      true,
      "never-prefix"
    ]
  },
  "extends": [
    "tslint:recommended",
    "tslint-react",
    "tslint-config-prettier"
  ],
  "linterOptions": {
    "exclude": [
      "config/**/*.js",
      "node_modules/**/*.ts",
      "coverage/lcov-report/*.js"
    ]
  }
}
```
What am I doing wrong?

**Comment by aaronjameslang** [comment]
@offg777 Nothing obvious. Link to repo?

**Comment by johnwiseheart** [comment]
The config looks correct to me - are you sure theres nothing else overriding it? The rule is definitely working properly for me

**Comment by stevenvanc** [comment]
@offg777 
> "interface name must start with a capitalized I"

From where do you get this error? From your IDE or when running a CLI?
And I don't know about the exact cascading rules of tslint, but have you tried putting the "rules" block underneath "extends"?

**Comment by DDDenial** [comment]
厉害

**Comment by ackvf** [comment]
Does eslint support that already?

**Comment by hyvyys** [comment]
Setting

```
"rules": {
    "interface-name": [
      true,
      "never-prefix"
    ]
  },
```

results in
`Configuration for rule "interface-name" is invalid`

Maybe this rule exists for a reason? Should every interface have a meaningful name obviously different from an implementing class?

**Comment by sergeych** [comment]
This is bloody annoying idea of forcing the rest of the world following your preferences _outside the your own project_. Obtrusion of your own naming conventions in programming tools is a bug of the product developer neural network ;)

**Comment by JoshuaKGoldberg** [comment]
Good thing TSLint is deprecated and this issue is no longer relevant, then! #4534
